src/App.tsx

```diff
@@ -1,8 +1,116 @@
+import { useState } from 'react';
+import { TextField, Box, Button } from '@mui/material';
 import './App.css';
 
 const App: React.FC = () => {
+  const [digits, setDigits] = useState<string[]>(new Array(8).fill(''));
+  const [lastDigit, setLastDigit] = useState<number | null>(null); 
+
+  const baseUrl: string = "http://localhost:3000";
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>, index: number) => {
+    const newDigits = [...digits];
+    const input = event.target.value;
+
+    if (input.match(/^\d$/)) {
+      newDigits[index] = input;
+      setDigits(newDigits);
+
+      if (input && index < digits.length - 1) {
+        const nextInput = document.getElementById(`digit-${index + 1}`) as HTMLInputElement;
+        if (nextInput) {
+          nextInput.focus();
+        }
+      }
+    } else {
+      event.preventDefault();
+    }
+  };
+
+  const handlePaste = (event: React.ClipboardEvent<HTMLInputElement>) => {
+    const pastedData = event.clipboardData.getData('Text').split('');
+    const newDigits = [...digits];
+
+    pastedData.forEach((char, i) => {
+      (i < digits.length && char.match(/^\d$/)) && (newDigits[i] = char);
+    });
+
+    setDigits(newDigits);
+    event.preventDefault();
+  };
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLDivElement>, index: number) => {
+    const newDigits = [...digits];
+    if (event.key === 'Backspace') {
+      if (digits[index] === '') {
+        if (index > 0) {
+          const prevInput = document.getElementById(`digit-${index - 1}`) as HTMLInputElement;
+          if (prevInput) {
+            prevInput.focus();
+          }
+        }
+      } else {
+        newDigits[index] = '';
+        setDigits(newDigits);
+      }
+    }
+  };
+
+  const getLastDigit = async () => {
+    const res = await fetch(`${baseUrl}/get`, {
+      method: "POST",
+      body: JSON.stringify({ "first8": `${digits.join("")}` }),
+      headers: {
+        'Content-Type': 'application/json',
+        'Accept': 'application/json'
+      }
+    });
+
+    return (await res.json()).lastDigit;
+  };
+
+  const handleClick = async () => {
+    await fetch(`${baseUrl}/save`, {
```

Comment:
you ar saving before knowing if its valid, u need to save after getting the result form server of the last digit

Reply:
the button cannot be clicked unless the input is valid (8 characters long, all digits).